src/hub/Hub.sol

```diff
@@ -458,16 +422,18 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler, IHubGuar
             IAccounting.AccountDoesNotExist()
         );
 
-        HoldingAccount[] memory accounts = new HoldingAccount[](4);
-        accounts[0] = HoldingAccount(assetAccount, uint8(AccountType.Asset));
-        accounts[1] = HoldingAccount(equityAccount, uint8(AccountType.Equity));
-        accounts[2] = HoldingAccount(gainAccount, uint8(AccountType.Gain));
-        accounts[3] = HoldingAccount(lossAccount, uint8(AccountType.Loss));
+        {
+            HoldingAccount[] memory accounts = new HoldingAccount[](4);
+            accounts[0] = HoldingAccount(assetAccount, uint8(AccountType.Asset));
+            accounts[1] = HoldingAccount(equityAccount, uint8(AccountType.Equity));
+            accounts[2] = HoldingAccount(gainAccount, uint8(AccountType.Gain));
+            accounts[3] = HoldingAccount(lossAccount, uint8(AccountType.Loss));
```

Comment:
Just a thought: Why don't we move the array initialization also to the hub helpers? IMO, this would align better with how we use the helper for other calls where the lower level logic is handled by the helper.

Reply:
Same for liabilities

Reply:
> Would you move just these 5 lines to `HubHelpers`, or the `initialize` call as well?
> 
> It seems to me we don't gain too much since you still need to pass all these params

Good point. I think I'd prefer moving the initialization to the helper then as well but no strong opinion - let's merge as is.